### ISSUE TITLE
Feature request: resolve a host name to an IP address

### ISSUE THREAD

**Issue by smcv** [python-zeroconf/python-zeroconf#129]
If a machine `mylaptop.local.` advertises a service `My Service._test._tcp.local.`, then I can:

* browse for all instances of `_test._tcp` in `.local.` and get the host name `mylaptop.local.` and the IP address as a side-effect (`examples/browser.py`)
* resolve the `My Service` instance of `_test._tcp.local.`, and get the host name `mylaptop.local.` and the IP address (#127 adds an example of this)

However, I can't see a straightforward way to ask "what is the address of `mylaptop.local.`?" through high-level APIs: combining `add_listener()` and `send()` seems to be the only way?

**Comment by pinkynrg** [comment]
@smcv would you be so nice to share your workaround?
I need to resolve a host name to an IP address and can't find a way to do it.

**Comment by pinkynrg** [comment]
right now I'm doing it using python sockets:

```
import socket
ip_address = socket.gethostbyname(host_name)
```

**Comment by smcv** [comment]
That will only work if your OS can resolve `.local` names using Zeroconf itself, for example a Linux system with [nss-mdns](https://github.com/lathiat/nss-mdns) installed. In the project I was thinking of when I opened this issue, we can't necessarily assume that.

**Comment by pinkynrg** [comment]
Yes I know and I would like to use add_listener + send. How did you do it?

**Comment by smcv** [comment]
Sorry, I don't have ready-to-use code for that. If I did, I'd have proposed it as new API.

It would be a lot like the part of `ServiceInfo.request` that sends `DNSQuestion(self.server, _TYPE_A, _CLASS_IN)`, and the part of `ServiceInfo.update_record` that responds to receiving `_TYPE_A` records.

**Comment by bdraco** [comment]
Its likely non-obvious that ServiceInfo can be subclassed to do this

example
https://github.com/esphome/esphome/blob/ea4e618f2af48b25c813ca9f54eddae530077c29/esphome/zeroconf.py#L19

**Comment by bdraco** [comment]
0.142.0 added an built in class and example
